doc/adr/0002-steppable-cek-machine.md

```diff
@@ -0,0 +1,235 @@
+# ADR 2: Steppable CEK machine
+
+Date: 2022-10
+
+## Authors
+
+Marty Stumpf <marty.stumpf@iohk.io>  
+Ziyang Liu <ziyang.liu@iohk.io>
+
+## Status
+
+Proposed
+
+## Context
+
+In order to have a minimal viable product of a debugger for Plutus, we need a CEK machine that will give us more information for debugging than our current one.
+
+Implementing the steppable CEK machine is a non-trivial task and involves some design decisions:
+
+- should the debugging machine be a separate one?
+- should we modify the existing CEK machine to support stepping, or implement something entirely new?
+- how do we ensure performance and conformance?
```

Comment:
This isn't a decision!

How about this:

We have a decision to make about whether or not to write the debuggable machine as a separate machine or try to share code with the production machine. The _requirements_ that we need to satisfy are a) we must not compromise the performance of the production machine, and b) the debugging machine must behave the same as the production machine.

Reply:
Plus, later on you say we're going to make a decision about how to architect the steppable machine also. As I said there I think that could be an ADR since we've not even got a prototype yet